### DISCUSSION TITLE
Characteristic Status Fault

### DISCUSSION THREAD

**Discussion by Sill1000** [HomeSpan/HomeSpan#578] [Q&A] [answered]
Hi,

If Characteristic Status Fault is set; the device is as inop or faulty.
But I don't see anything changing in the Homekit app (the last value is still displayed). Is it hidden somewhere else?

Thanks

**Comment by HomeSpan** [comment]
The fault indictor is usually found within the each Accessory Tile's "Settings" page.  Apple does not display this on the main pages of the Home App (though it will show a battery low warning icon directly on the Tile, so I don't know why they did not do the same for a fault indicator.

**Comment by Sill1000** [comment]
Fault is not available for all sensor but tried to:

Set Fault = 1 on a temperature sensor. Nothing on the app.
![Copie de IMG_6487](https://github.com/HomeSpan/HomeSpan/assets/88202590/71a522a1-3530-423e-96cf-404f8576cb13)

**Comment by HomeSpan** [comment, marked as answer]
I double-checked and can confirm that Status Fault no longer seems to be displayed in the Home App, though it is still displayed properly in the _Eve for HomeKit_ App (suggesting that the Characteristic is properly configured within HomeSpan).  I guess Apple deprecated the Characteristic.  Unfortunately Apple no longer provides any API documentation for HAP so changes like this can only be identified by trial and error.

```C++
  new SpanAccessory();
    new Service::AccessoryInformation();
      new Characteristic::Identify();
    new Service::TemperatureSensor();
      new Characteristic::Name("OutdoorTemp");        // dislpayed correctly
      new Characteristic::CurrentTemperature(20);     // displayed correctly
      new Characteristic::StatusActive(true);         // displayed correctly (in settings screen only)
      new Characteristic::StatusFault(1);             // NOT DISPLAYED IN HOME APP, BUT DISPLAYED IN EVE
      new Characteristic::StatusLowBattery(0);        // displayed correctly (in settings screen only)
      new Characteristic::StatusTampered(0);          // displayed correctly (in settings screen only
```

**Reply by Sill1000** [reply]
Good to know!!! Thanks !!!

**Reply by frankonski** [reply]
Something interesting I just found: I can confirm that with iOS 16.7, I see the same results as @HomeSpan mentioned above, which is that there is no indication whatsoever of a StatusFault in the accessory. Nothing in the accessory's settings, no icon, nothing.

I discovered the _Controller for HomeKit_ app today, and this app **does** have a line that shows the StatusFault value for that same accessory.

This tells me that iOS still stores that characteristic but unfortunately does nothing with it.
